### PR TITLE
Migrated Mcp960x from .NET IoT to .NET

### PR DESCRIPTION
**Pull request by mork2020**
## Description

- migrated device Mcp960x from .NET IoT to .NET
- the major changes are the replacement of the `System.Convert.ToBoolean` method in `Mcp960x.cs` line 123 to 130 with `!= 0` 

## Motivation and Context

- Resolves nanoFramework/Home#857

## How Has This Been Tested?

- i tested the changes with an ESP32 board and a MCP9600 breakout board

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by josesimoes** [comment]
@mork2020 thanks for tackling this one! 👍🏻 
Please check the build log here https://dev.azure.com/nanoframework/IoT.Device/_build/results?buildId=28853&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=d1e88c09-dceb-53c1-5579-19c846f3526a

**Review by Ellerbach — COMMENTED** [review]
Thanks a lot! Looks great.
Can you adjust the fritzing and the exported image to have a nano device and not a RPI?

File: devices/Mcp960x/Mcp960x.sln [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Visual Studio Version 17
+VisualStudioVersion = 17.1.32210.238
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mcp960x", "Mcp960x.nfproj", "{B16D750D-0001-48F9-863D-FD876E8C79A7}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mcp960x.Samples", "Samples\Mcp960x.Samples.nfproj", "{7745867C-1E31-162E-E8A0-9E65A6327349}"
+EndProject
+Global
+	GlobalSection(SharedMSBuildProjectFiles) = preSolution
+		..\..\src\System.Runtime.CompilerService\System.Runtime.CompilerService.projitems*{b16d750d-0001-48f9-863d-fd876e8c79a7}*SharedItemsImports = 5
```
[the remark this answers is not in the document]
Comment (mork2020) [reply devices/Mcp960x/Mcp960x.sln]: reference removed

File: devices/Mcp960x/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+```csharp
+//////////////////////////////////////////////////////////////////////
+// when connecting to an ESP32 device, need to configure the I2C GPIOs
+// used for the bus
+Configuration.SetPinFunction(21, DeviceFunction.I2C1_DATA);
+Configuration.SetPinFunction(22, DeviceFunction.I2C1_CLOCK);
+```
+
+For other devices like STM32, please make sure you're using the preset pins for the I2C bus you want to use.
+
```
[the remark this answers is not in the document]
Comment (mork2020) [reply devices/Mcp960x/README.md]: line removed

**Comment by mork2020** [comment]
This device isn't acutally automated imported from the .NET IoT repository because the System.Convert.ToBoolean method is missing in the nanoframework. Should we create a PR for the .NET IoT repository with the changes in Mcp960x.cs line 123 to 130?

**Comment by Ellerbach** [comment]
> System.Convert.ToBoolean method is missing in the nanoframework

Yes, but it's just 1 line to adjust.

> Should we create a PR for the .NET IoT repository with the changes in Mcp960x.cs line 123 to 130

No, it's not needed, there are changes between both because of the Span<byte> and few other elements like this or enums comparison. Some have much more.

**Review by Ellerbach — APPROVED** [review]
Thanks a lot @mork2020 and I ❤️ the real picture!

**Pull request merged**